sbin/create_course.sh

```diff
@@ -14,56 +14,56 @@ fi
 
 CONF_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"/../config
 
-SUBMITTY_REPOSITORY_DIR=$(jq -r '.submitty_repository' ${CONF_DIR}/submitty.json)
-SUBMITTY_INSTALL_DIR=$(jq -r '.submitty_install_dir' ${CONF_DIR}/submitty.json)
-SUBMITTY_DATA_DIR=$(jq -r '.submitty_data_dir' ${CONF_DIR}/submitty.json)
-SUBMISSION_URL=$(jq -r '.submission_url' ${CONF_DIR}/submitty.json)
+SUBMITTY_REPOSITORY_DIR=$(jq -r '.submitty_repository' "${CONF_DIR}/submitty.json")
+SUBMITTY_INSTALL_DIR=$(jq -r '.submitty_install_dir' "${CONF_DIR}/submitty.json")
+SUBMITTY_DATA_DIR=$(jq -r '.submitty_data_dir' "${CONF_DIR}/submitty.json")
+SUBMISSION_URL=$(jq -r '.submission_url' "${CONF_DIR}/submitty.json")
 
-PHP_USER=$(jq -r '.php_user' ${CONF_DIR}/submitty_users.json)
-DAEMON_USER=$(jq -r '.daemon_user' ${CONF_DIR}/submitty_users.json)
-CGI_USER=$(jq -r '.cgi_user' ${CONF_DIR}/submitty_users.json)
+PHP_USER=$(jq -r '.php_user' "${CONF_DIR}/submitty_users.json")
+DAEMON_USER=$(jq -r '.daemon_user' "${CONF_DIR}/submitty_users.json")
+CGI_USER=$(jq -r '.cgi_user' "${CONF_DIR}/submitty_users.json")
 
-COURSE_BUILDERS_GROUP=$(jq -r '.course_builders_group' ${CONF_DIR}/submitty_users.json)
+COURSE_BUILDERS_GROUP=$(jq -r '.course_builders_group' "${CONF_DIR}/submitty_users.json")
 
-DATABASE_HOST=$(jq -r '.database_host' ${CONF_DIR}/database.json)
-DATABASE_PORT=$(jq -r '.database_port' ${CONF_DIR}/database.json)
-DATABASE_USER=$(jq -r '.database_user' ${CONF_DIR}/database.json)
-DATABASE_PASS=$(jq -r '.database_password' ${CONF_DIR}/database.json)
+DATABASE_HOST=$(jq -r '.database_host' "${CONF_DIR}/database.json")
+DATABASE_PORT=$(jq -r '.database_port' "${CONF_DIR}/database.json")
+DATABASE_USER=$(jq -r '.database_user' "${CONF_DIR}/database.json")
+DATABASE_PASS=$(jq -r '.database_password' "${CONF_DIR}/database.json")
 
 ########################################################################################################################
 ########################################################################################################################
 
 CONN_STRING="-h ${DATABASE_HOST} -U ${DATABASE_USER}"
-if [ -d ${DATABASE_HOST} ]; then
+if [ -d "${DATABASE_HOST}" ]; then
     CONN_EXTRA="${CONN_STRING} -p ${DATABASE_PORT}"
```

Comment:
`${CONN_EXTRA}` is never used...

Reply:
Same as with the instances of SC2006 in this file, I'll come back through with a second PR (hopefully when I add shellcheck linting) to fix more complicated things like that.  I think `CONN_EXTRA` is safe to remove here, but it can be risky to remove things like that without being very thorough in reading the code.  I specifically wanted to be strict in only quoting variables in this PR, making only the most minimal edits possible to minimize the risk of breakages.

I'm not opposed to the idea of going ahead and fully linting the entire file, but it can be a little more risky.  Thoughts on this?

Reply:
Actually, nowhere in this repository reads `CONN_EXTRA`... 

Maybe it could be safely deleted in your second PR if the CI passes?